### DISCUSSION TITLE
Any way to get substrings?

### DISCUSSION THREAD

**Discussion by Fran314** [elkowar/eww#811] [Q&A] [answered]
I find that expressions in yuck are sometimes a bit limited. As far as I know there is no way to get a substring from a string, which would actually be pretty handy to handle text in a top bar, which could be truncated and then eventually expanded with a revealer. I guess I could obtain a substring with the replace function, but that seems clunky. Is there really no way to get a substring from a string?

**Comment by mantacid** [comment]
i think you could use the search(string, regex) function with a regex pattern to lookbehind for the first X characters, and match the subsequent Y characters, to get the substring of length Y that starts at X. but good luck trying to get regex to work.

**Reply by mantacid** [reply]
I figured it out. Use search as I had said, and enclose the regex in single quotes. Idk if that last part is important, but it is if you call the regex inside a string or something. Just remember that this function returns an array, so you’ll need to specify an index with square brackets after the search function is called.

**Comment by elkowar** [comment, marked as answer]
I added a substring function to simplexpr now, update eww and check the docs ^^
